platform/openide.dialogs/src/org/openide/NotifyDescriptor.java

```diff
@@ -129,24 +129,36 @@ public class NotifyDescriptor extends Object {
      */
     public static final String PROP_INFO_NOTIFICATION = "infoNotification"; // NOI18N
 
+    // Options used to be represented as internal Integer instances with distinct
+    // object identities.
+    // Since YES and OK options have the same int value, Integer.valueOf()
+    // can't be used here.
+    private enum OPTIONS {
+        YES_OPTION,
+        NO_OPTION,
+        CANCEL_OPTION,
+        OK_OPTION,
+        CLOSED_OPTION;
+    }
+    
     //
     // Return values
     //
 
     /** Return value if YES is chosen. */
-    public static final Object YES_OPTION = new Integer(JOptionPane.YES_OPTION);
+    public static final Object YES_OPTION = OPTIONS.YES_OPTION;
 
     /** Return value if NO is chosen. */
-    public static final Object NO_OPTION = new Integer(JOptionPane.NO_OPTION);
+    public static final Object NO_OPTION = OPTIONS.NO_OPTION;
 
     /** Return value if CANCEL is chosen. */
-    public static final Object CANCEL_OPTION = new Integer(JOptionPane.CANCEL_OPTION);
+    public static final Object CANCEL_OPTION = OPTIONS.CANCEL_OPTION;
 
     /** Return value if OK is chosen. */
-    public static final Object OK_OPTION = new Integer(JOptionPane.OK_OPTION);
+    public static final Object OK_OPTION = OPTIONS.OK_OPTION;
 
     /** Return value if user closes the window without pressing any button. */
-    public static final Object CLOSED_OPTION = new Integer(JOptionPane.CLOSED_OPTION);
+    public static final Object CLOSED_OPTION = OPTIONS.CLOSED_OPTION;
```

Comment:
we could also simply use `new Object()` for all of them. The fact that they were internally based on the values of `JOptionPane` was just an implementation detail and shouldn't really matter externally.

Reply:
looks like we will have to replicate Integer completely